### PR TITLE
daemon: Deprecate KPR=probe

### PR DESCRIPTION
**Pull request by brb**
The option is a never ending source of confusion. Also, after changing
system configuration, the "probe" option could unexpectedly enable some
KPR features which might be not expected by users.

Deprecate the "probe" option for now. In v1.13 we will make the option
to switch to "disabled". Unfortunately, we cannot just remove the
option, as it was used as a default in the Helm charts.

### CODE REVIEW

File: install/kubernetes/cilium/templates/cilium-configmap.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@
 {{- $enableIdentityMark := "true" -}}
 {{- $fragmentTracking := "true" -}}
 {{- $crdWaitTimeout := "5m" -}}
-{{- $defaultKubeProxyReplacement := "probe" -}}
```
Comment (aanm) [inline install/kubernetes/cilium/templates/cilium-configmap.yaml:16]: This needs a similar logic as done on lines 20-48 to avoid breaking upgrades.

**Review by aanm — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]
Looks good!

**Review by borkmann — APPROVED** [review]
lgtm

**Pull request merged**